### PR TITLE
Storage-Image

### PR DESCRIPTION
**Pull request by QuocHuy178**
Testing Result:
image

### CODE REVIEW

File: src/Services/GTT/GTT.Api/Configuration/Routes.cs [hunk 1]
```diff
@@ -29,6 +29,14 @@ public static class Routes
     /// </summary>
     #region ExerciseManagement
     public const string ExerciseLibrary = "v1/excercise-library";
+    public const string ExerciseLibraryImage = "v1/excercise-library/upload-image";
```
Comment (ClearActionConsistent) [inline src/Services/GTT/GTT.Api/Configuration/Routes.cs]: Why do we need both, ExerciseLibraryImage and UploadImage(line 40).
I think the route name UploadImage is more descriptive

File: src/Services/GTT/GTT.Api/local.settings.json [hunk 2]
```diff
@@ -3,7 +3,9 @@
   "Values": {
     "AzureWebJobsStorage": "UseDevelopmentStorage=true",
     "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",
-    "SqlOptions__SqlConnectionString": "Data Source=tcp:alexdev.database.windows.net;Initial Catalog=alex;Persist Security Info=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;User ID=ffffff;Password=ffffff;",
-    "SqlOptions__SqlUseAccessToken": "false"
+    "SqlOptions__SqlConnectionString": "Data Source=VPHUDUC;Initial Catalog= GTT; database = GTT; User ID=sa;Password=123456",
+    "SqlOptions__SqlUseAccessToken": "false",
+    "AzureWebJobsStorageImage": "fffff",
```
Comment (ClearActionConsistent) [inline src/Services/GTT/GTT.Api/local.settings.json]: Can we group options for uploading image using prefix like UploadImageOptions__properties...something like SqlOptions

File: src/Services/GTT/shared/GTT.Application/Commands/UploadImageStorage.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+                if (!file.ContentType.Contains(Constants.AzureStorageImage.Type))
+                {
+                    return new BaseResponseModel(HttpStatusCode.InternalServerError, "File Type Should Be Image");
+                }
+
+                var data = file.Data;
+                if (ConvertBytesToMegabytes(data.Length) > Constants.AzureStorageImage.Size)
+                {
+                    return new BaseResponseModel(HttpStatusCode.InternalServerError, "File Size Too Large, Must Be Less Than 10MB");
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Commands/UploadImageStorage.cs]: "10MB" should be set dynamically based on Constants.AzureStorageImage.Size, BTW it should be 1MB instead

File: src/Services/GTT/shared/GTT.Application/Commands/UploadImageStorage.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+                var blob = blobClient.GetBlobClient(file.FileName);
+                await blob.UploadAsync(myBlob);
+
+                var result = new UploadImage { ImageUrl = blob.Uri.AbsoluteUri };
+
+                return new BaseResponseModel(result);
+            }
+
+            static double ConvertBytesToMegabytes(long bytes)
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Commands/UploadImageStorage.cs]: This is a private function, just make it private

File: src/Services/GTT/shared/GTT.Application/Requests/MultiPartFormDataModel.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GTT.Application.Requests
+{
+    public class MultiPartFormDataModel
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Requests/MultiPartFormDataModel.cs]: Class name should be more descriptive, UploadedImageData I think
Missing field UserId

File: src/Services/GTT/shared/GTT.Application/Utils/Constants.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+using System.Threading.Tasks;
+
+namespace GTT.Application.Utils
+{
+    public class Constants
+    {
+        public class AzureStorageImage
+        {
+            public const string Type = "image";
+            public const int Size = 10;
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Utils/Constants.cs]: Size should be more descriptive, change to UploadedImageFileSize
Same to Type, if Type is for valid uploaded image file types, than change the variable name to : AllowUploadedImageFileTypes. And value should be something like ".jpg, .png, .gif, .jpeg" for now

File: src/Services/GTT/shared/GTT.Domain/Entities/UploadImage.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GTT.Domain.Entities
+{
+    public class UploadImage
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Domain/Entities/UploadImage.cs]: Objects in GTT.Domain represents for tables in DB, this class is not for that purpose, so move it to GTT.Application

**Review by phantuananh18 — APPROVED** [review]

**Review by phantuananh18 — APPROVED** [review]

**Pull request merged**